alacritty/src/cli.rs

```diff
@@ -169,6 +171,7 @@ impl Options {
                         .allow_hyphen_values(true)
                         .help("Command and args to execute (must be last argument)"),
                 )
+                .arg(Arg::with_name("hold").long("hold").help("Remain open after child process exits"))
```

Comment:
This formatting seems odd. You should format your code with `cargo +nightly fmt` (p.s. nightly branch is required for alacritty atm).

Reply:
@kchibisov Hmm, interesting. When I do `cargo + nightly fmt` in the project root, it reformats a `let matches` section completely, but it doesn't touch my `hold` match addition. I'm not sure if I'm doing it right.

```patch
diff --git a/alacritty/src/cli.rs b/alacritty/src/cli.rs
index d1bd919..c697850 100644
--- a/alacritty/src/cli.rs
+++ b/alacritty/src/cli.rs
@@ -74,105 +74,106 @@ impl Options {
 
         let mut options = Options::default();
 
-        let matches =
-            App::new(crate_name!())
-                .version(version.as_str())
-                .author(crate_authors!("\n"))
-                .about(crate_description!())
-                .arg(Arg::with_name("ref-test").long("ref-test").help("Generates ref test"))
-                .arg(
-                    Arg::with_name("live-config-reload")
-                        .long("live-config-reload")
-                        .help("Enable automatic config reloading"),
-                )
-                .arg(
-                    Arg::with_name("no-live-config-reload")
-                        .long("no-live-config-reload")
-                        .help("Disable automatic config reloading")
-                        .conflicts_with("live-config-reload"),
-                )
-                .arg(
-                    Arg::with_name("print-events")
-                        .long("print-events")
-                        .help("Print all events to stdout"),
-                )
-                .arg(
-                    Arg::with_name("persistent-logging")
-                        .long("persistent-logging")
-                        .help("Keep the log file after quitting Alacritty"),
-                )
-                .arg(
-                    Arg::with_name("dimensions")
-                        .long("dimensions")
-                        .short("d")
-                        .value_names(&["columns", "lines"])
-                        .help(
-                            "Defines the window dimensions. Falls back to size specified by \
-                             window manager if set to 0x0 [default: 0x0]",
-                        ),
-                )
-                .arg(
-                    Arg::with_name("position")
-                        .long("position")
-                        .allow_hyphen_values(true)
-                        .value_names(&["x-pos", "y-pos"])
-                        .help(
-                            "Defines the window position. Falls back to position specified by \
-                             window manager if unset [default: unset]",
-                        ),
-                )
-                .arg(
-                    Arg::with_name("title")
-                        .long("title")
-                        .short("t")
-                        .takes_value(true)
-                        .help(&format!("Defines the window title [default: {}]", DEFAULT_NAME)),
-                )
-                .arg(
-                    Arg::with_name("class").long("class").takes_value(true).help(&format!(
-                        "Defines window class on Linux [default: {}]",
-                        DEFAULT_NAME
-                    )),
-                )
-                .arg(Arg::with_name("embed").long("embed").takes_value(true).help(
+        let matches = App::new(crate_name!())
+            .version(version.as_str())
+            .author(crate_authors!("\n"))
+            .about(crate_description!())
+            .arg(Arg::with_name("ref-test").long("ref-test").help("Generates ref test"))
+            .arg(
+                Arg::with_name("live-config-reload")
+                    .long("live-config-reload")
+                    .help("Enable automatic config reloading"),
+            )
+            .arg(
+                Arg::with_name("no-live-config-reload")
+                    .long("no-live-config-reload")
+                    .help("Disable automatic config reloading")
+                    .conflicts_with("live-config-reload"),
+            )
+            .arg(
+                Arg::with_name("print-events")
+                    .long("print-events")
+                    .help("Print all events to stdout"),
+            )
+            .arg(
+                Arg::with_name("persistent-logging")
+                    .long("persistent-logging")
+                    .help("Keep the log file after quitting Alacritty"),
+            )
+            .arg(
+                Arg::with_name("dimensions")
+                    .long("dimensions")
+                    .short("d")
+                    .value_names(&["columns", "lines"])
+                    .help(
+                        "Defines the window dimensions. Falls back to size specified by window \
+                         manager if set to 0x0 [default: 0x0]",
+                    ),
+            )
+            .arg(
+                Arg::with_name("position")
+                    .long("position")
+                    .allow_hyphen_values(true)
+                    .value_names(&["x-pos", "y-pos"])
+                    .help(
+                        "Defines the window position. Falls back to position specified by window \
+                         manager if unset [default: unset]",
+                    ),
+            )
+            .arg(
+                Arg::with_name("title")
+                    .long("title")
+                    .short("t")
+                    .takes_value(true)
+                    .help(&format!("Defines the window title [default: {}]", DEFAULT_NAME)),
+            )
+            .arg(
+                Arg::with_name("class")
+                    .long("class")
+                    .takes_value(true)
+                    .help(&format!("Defines window class on Linux [default: {}]", DEFAULT_NAME)),
+            )
+            .arg(
+                Arg::with_name("embed").long("embed").takes_value(true).help(
                     "Defines the X11 window ID (as a decimal integer) to embed Alacritty within",
-                ))
-                .arg(
-                    Arg::with_name("q")
-                        .short("q")
-                        .multiple(true)
-                        .conflicts_with("v")
-                        .help("Reduces the level of verbosity (the min level is -qq)"),
-                )
-                .arg(
-                    Arg::with_name("v")
-                        .short("v")
-                        .multiple(true)
-                        .conflicts_with("q")
-                        .help("Increases the level of verbosity (the max level is -vvv)"),
-                )
-                .arg(
-                    Arg::with_name("working-directory")
-                        .long("working-directory")
-                        .takes_value(true)
-                        .help("Start the shell in the specified working directory"),
-                )
-                .arg(Arg::with_name("config-file").long("config-file").takes_value(true).help(
-                    "Specify alternative configuration file [default: \
-                     $XDG_CONFIG_HOME/alacritty/alacritty.yml]",
-                ))
-                .arg(
-                    Arg::with_name("command")
-                        .long("command")
-                        .short("e")
-                        .multiple(true)
-                        .takes_value(true)
-                        .min_values(1)
-                        .allow_hyphen_values(true)
-                        .help("Command and args to execute (must be last argument)"),
-                )
-                .arg(Arg::with_name("hold").long("hold").help("Remain open after child process exits"))
-                .get_matches();
+                ),
+            )
+            .arg(
+                Arg::with_name("q")
+                    .short("q")
+                    .multiple(true)
+                    .conflicts_with("v")
+                    .help("Reduces the level of verbosity (the min level is -qq)"),
+            )
+            .arg(
+                Arg::with_name("v")
+                    .short("v")
+                    .multiple(true)
+                    .conflicts_with("q")
+                    .help("Increases the level of verbosity (the max level is -vvv)"),
+            )
+            .arg(
+                Arg::with_name("working-directory")
+                    .long("working-directory")
+                    .takes_value(true)
+                    .help("Start the shell in the specified working directory"),
+            )
+            .arg(Arg::with_name("config-file").long("config-file").takes_value(true).help(
+                "Specify alternative configuration file [default: \
+                 $XDG_CONFIG_HOME/alacritty/alacritty.yml]",
+            ))
+            .arg(
+                Arg::with_name("command")
+                    .long("command")
+                    .short("e")
+                    .multiple(true)
+                    .takes_value(true)
+                    .min_values(1)
+                    .allow_hyphen_values(true)
+                    .help("Command and args to execute (must be last argument)"),
+            )
+            .arg(Arg::with_name("hold").long("hold").help("Remain open after child process exits"))
+            .get_matches();
 
         if matches.is_present("ref-test") {
             options.ref_test = true;
diff --git a/alacritty_terminal/src/term/mod.rs b/alacritty_terminal/src/term/mod.rs
index d2014ce..f073174 100644
--- a/alacritty_terminal/src/term/mod.rs
+++ b/alacritty_terminal/src/term/mod.rs
@@ -2340,12 +2340,9 @@ mod tests {
         };
         let mut config = MockConfig::default();
         config.hold = true;
-        let mut term = Term::new(
-            &config,
-            &size,
-            Clipboard::new_nop(),
-            EventSpy { latest_event: RefCell::new(None) },
-        );
+        let mut term = Term::new(&config, &size, Clipboard::new_nop(), EventSpy {
+            latest_event: RefCell::new(None),
+        });
         term.handle_child_process_exit();
         assert_eq!(term.event_proxy.latest_event.into_inner(), None);
     }
@@ -2363,12 +2360,9 @@ mod tests {
         };
         let mut config = MockConfig::default();
         config.hold = false;
-        let mut term = Term::new(
-            &config,
-            &size,
-            Clipboard::new_nop(),
-            EventSpy { latest_event: RefCell::new(None) },
-        );
+        let mut term = Term::new(&config, &size, Clipboard::new_nop(), EventSpy {
+            latest_event: RefCell::new(None),
+        });
         term.handle_child_process_exit();
         assert_eq!(term.event_proxy.latest_event.into_inner(), Some(Event::Exit));
     }
```

Reply:
`config-file` arg formatting looks kinda similar.

Reply:
Anyways, I reformatted it in the same ways as most of the args formatted, I hope it's good enough now.